pkg/graph/resolvers/system_intake_grb_review_form.go

```diff
@@ -100,3 +102,34 @@ func UpdateSystemIntakeGRBReviewFormInputTimeframeAsync(
 		SystemIntake: updatedIntake,
 	}, nil
 }
+
+// PointerToSystemIntakeGRBReviewAsyncStatusType Create a reference to an Async GRB Review status
+func PointerToSystemIntakeGRBReviewAsyncStatusType(
+	status models.SystemIntakeGRBReviewAsyncStatusType,
+) *models.SystemIntakeGRBReviewAsyncStatusType {
+	return &status
+}
+
+// CalcSystemIntakeGRBReviewAsyncStatus calculates the status of the GRB Review Async page
+func CalcSystemIntakeGRBReviewAsyncStatus(
+	intake *models.SystemIntake,
+) (*models.SystemIntakeGRBReviewAsyncStatusType, error) {
+	currentTime := time.Now()
+
+	if intake.GrbReviewType != models.SystemIntakeGRBReviewTypeAsync {
+		return nil, errors.New("system intake is not of type GRB Review Async")
+	}
+
+	if intake.GrbReviewAsyncEndDate == nil {
+		return nil, errors.New("system intake GRB Review Async end date is not set")
+	}
+
+	// Evaluate if the current time is before the Grb Review Async end date
+	if intake.GrbReviewAsyncEndDate.After(currentTime) {
+		return PointerToSystemIntakeGRBReviewAsyncStatusType(models.SystemIntakeGRBReviewAsyncStatusTypeInProgress), nil
```

Comment:
Here is an example of using the helper method.
```suggestion
return helpers.PointerTo(models.SystemIntakeGRBReviewAsyncStatusTypeInProgress), nil
```